File: packages/core/tests/fixtures/unhealthy/complex.rs
fn process_data(user: &User, config: &Config, logger: &Logger, db: &Database, cache: &Cache) -> Result<Vec<Item>, Error> {
    if user.is_active() {
        for item in db.get_items()? {
            match item.status {
                Status::Active => {
                    if item.priority > 10 {
                        while cache.is_locked() {
                            std::thread::sleep(Duration::from_millis(100));
                        }
                        logger.log(&format!("Processing {}", item.id));
                    }
                }
                Status::Pending => {
                    if item.created_at < config.cutoff {
                        return Err(Error::TooOld);
                    }
                }
                _ => {}
            }
        }
    }
    Ok(vec![])
}

File: packages/core/tests/fixtures/unhealthy/complex.swift
func processData(user: User, config: Config, logger: Logger, db: Database, cache: Cache) throws -> [Item] {
    var result: [Item] = []
    if user.isActive {
        for item in db.getItems() {
            switch item.status {
            case .active:
                if item.priority > 10 {
                    while cache.isLocked {
                        Thread.sleep(forTimeInterval: 0.1)
                    }
                    logger.log(item.id)
                    result.append(item)
                }
            case .pending:
                if item.createdAt < config.cutoff {
                    throw AppError.tooOld
                }
            default:
                break
            }
        }
    }
    return result
}
